Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/RoslynTargetsTests.cs

```diff
@@ -277,6 +277,59 @@ public void Roslyn_Settings_ValidSetup_NonLegacyServer_MergeSettings()
             AssertWarningsAreNotTreatedAsErrorsNorIgnored(result);
         }
 
+        [TestMethod]
+        public void Roslyn_Settings_LanguageMissing_NoError()
+        {
+            // Arrange
+
+            // Set the config directory so the targets know where to look for the analysis config file
+            var confDir = TestUtils.CreateTestSpecificFolder(TestContext, "config");
+
+            // Create a valid config file that does not contain analyzer settings
+            var config = new AnalysisConfig();
+            var configFilePath = Path.Combine(confDir, FileConstants.ConfigFileName);
+            config.Save(configFilePath);
+
+            // Create the project
+            var properties = new WellKnownProjectProperties
+            {
+                SonarQubeConfigPath = confDir,
+                ResolvedCodeAnalysisRuleset = "c:\\should.be.overridden.ruleset"
+            };
+
+            var projectRoot = CreateValidProjectSetup(properties);
+            projectRoot.AddProperty("Language", "");
+
+            projectRoot.AddItem(TargetProperties.AnalyzerItemType, "should.be.removed.analyzer1.dll");
+            const string additionalFileName = "should.not.be.removed.additional1.txt";
+            projectRoot.AddItem(TargetProperties.AdditionalFilesItemType, additionalFileName);
+
+            projectRoot.Save(); // re-save the modified project
+
+            // Act
+            var result = BuildRunner.BuildTargets(TestContext, projectRoot.FullPath, TargetConstants.OverrideRoslynAnalysisTarget);
+
+            var projectSpecificConfFilePath = result.GetCapturedPropertyValue(TargetProperties.ProjectConfFilePath);
+
+            var expectedRoslynAdditionalFiles = new string[] {
+                projectSpecificConfFilePath,
+                additionalFileName /* additional files are not removed */
+            };
+
+            // Assert
+            result.AssertTargetExecuted(TargetConstants.OverrideRoslynAnalysisTarget);
+            result.AssertTargetExecuted(TargetConstants.SetRoslynAnalysisPropertiesTarget);
+            result.BuildSucceeded.Should().BeTrue();
+
+            // Check the error log and ruleset properties are set
+            result.MessageLog.Contains("Analysis language is not specified");
```

Comment:
... check the expected message is in the log.

Reply:
It would be worth to either rename `Check the error log` into `Check the log for message` or something like that. Because reading the comment, I expect to have a failing build (to actually find the error message `Analysis language is not specified`).